docs/website/docs/developers/general/testing-guide.md

```diff
@@ -400,3 +400,84 @@ our `CMakeLists.txt` file by
 There are other test targets that generate tests based on template configuraton
 and platform detection, such as `iree_static_linker_test`. Those targets are
 not supported by Bazel rules at this point.
+
+## External test suite
+
+An out-of-tree test suite is under development at
+[nod-ai/SHARK-TestSuite](https://github.com/nod-ai/SHARK-TestSuite/tree/main/iree_tests)
+for large collections of generated tests and machine learning models that are
+too large to fit into the main git repository.
+
+Testing these programs follows several stages:
+
+```mermaid
+graph LR
+  Import -. "\n(offline)" .-> Compile
+  Compile --> Run
+```
+
+This particular test suite treats importing (e.g. from ONNX, PyTorch, or
+TensorFlow) as an offline step and contains test cases organized into folders
+of programs, inputs, and expected outputs:
+
+```text title="Sample test case directory"
+test_case_name/
+  model.mlir
+  input_0.npy
+  output_0.npy
+  test_data_flags.txt
+```
+
+```text title="Sample test_data_flags.txt"
+--input=@input_0.npy
+--expected_output=@output_0.npy
+```
+
+* Many model, input, and output files are too large to store directly in Git,
+so the external test suite also uses [Git LFS](https://git-lfs.com/) and cloud
+storage.
+
+Each test case can be run using a sequence of commands like:
+
+```bash
+iree-compile model.mlir {flags} -o model.vmfb
+iree-run-module --module=model.vmfb --flagfile=test_data_flags.txt
+```
+
+To run slices of the test suite, a [pytest](https://docs.pytest.org/en/stable/)
+runner is included that can be configured using JSON files. The JSON files
+tested in the IREE repo itself are stored in
+[`build_tools/pkgci/external_test_suite/`](https://github.com/openxla/iree/tree/main/build_tools/pkgci/external_test_suite).
+
+For example, here is part of a config file for running ONNX tests on CPU:
+
+```json title="build_tools/pkgci/external_test_suite/onnx_cpu_llvm_sync.json" linenums="1"
+--8<-- "build_tools/pkgci/external_test_suite/onnx_cpu_llvm_sync.json::20"
```

Comment:
Is this formatting intentional ?

Reply:
Yep! magic for embedding a file in a code block (up to line 20), and using a code block title: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#embedding-external-files

![image](https://github.com/openxla/iree/assets/4010439/65111e16-6212-4477-81fc-1af6e8a6f6fd)